Requisitos de la implementación

nepito opened the pull request:
# Requisitos
## [Implementación](https://github.com/IslasGECI/manual#implementaci%C3%B3n-en-python)
- [x] Evoca mutantes de Python con Mutmut en GitHub Actions
- [x] Corre una prueba dummy con Pytest en GitHub Actions
- [x] Evalúa cambio de cobertura con Codecov en GitHub Actions
- [x] Verifica formato de Python con Black en GitHub Actions
- [x] Agrega phony lint a Makefile para correr Flake8 y Pylint (localmente, no en GitHub Actions)

# Revisiones
## [1ra revisión](https://github.com/IslasGECI/mean_differences_mc/pull/2#pullrequestreview-476956146)
- [x] [Las clases deben ser nombradas como CamelCase segun PEP-8](https://github.com/IslasGECI/mean_differences_mc/pull/2#discussion_r478597692)
- [x] [Parece que no usamos la variable `i_switch`](https://github.com/IslasGECI/mean_differences_mc/pull/2#discussion_r478599618)
- [x] [En este archivo, los nombres de las variables deberían ser `a_final`, `b_final`, `A` y `B` para cumplir con PEP-8.](https://github.com/IslasGECI/mean_differences_mc/pull/2#discussion_r478604451)
---

- [**GECI-62**: Prueba diferencias entre medias utilizando técnicas Monte-Carlo](https://app.gitkraken.com/glo/card/85f658785b9a4fbeaeb6970209e2bbe0)

Commit:
Agrega la paquetería necesaria para el calentamiento

Commit:
Agrega los objetivos del calentamiento

Commit:
Da formato con black

Commit:
Agrega actions canónico

Commit:
Actualiza nombre de objetivo

Commit:
Cambia de unittest a pytest

Commit:
Cambia de unittest a pytest

Commit:
Actualiza los índices que cambiará

Commit:
Cambia a pytest

Commit:
Da formato utilizando black

GenaroVel committed:
Agrega tabla del ejemplo 4.2 de Chapman

GenaroVel committed:
Cambia estructura de los datos

Commit:
Prueba que calcula la diferencia de los promedios

Commit:
Pasa la primer prueba

Commit:
Pasa la segunda prueba

Commit:
Da formato con black

GenaroVel committed:
Agrega encabezados a tabla de ejemplo

Commit:
prueba que haga una lista de diferencias

Commit:
Quita dependencia obsoleta

Commit:
Crea función que genera un arreglo de diferencias

Commit:
Prueba el p valor

Commit:
Refactoriza la variable

Commit:
Pasa la prueba de la manera más absurda posible

Commit:
Agrega rope para refactorizar

Commit:
agrega setup.py

Commit:
agrega modulo para calcular p-valor

Commit:
agrega prueba del modulo para calcular p-valor

devarops committed:
Da formato usando Black

devarops commented on mean_differences_mc/Tester_Significant_Difference.py:
```diff
@@ -1,6 +1,7 @@
 from .index_switch import index_to_switch
 
+
 class Tester_Significant_Difference:
```
Las clases deben ser nombradas como CamelCase segun [PEP-8](https://www.python.org/dev/peps/pep-0008/#class-names)

devarops commented on mean_differences_mc/calculate_mean_difference.py:
```diff
@@ -0,0 +1,30 @@
+    minuend_switched = minuend
+    subtrahend_switched = subtrahend
+    mean_switched_difference = []
+    for i_switch in range(switches):
```
Parece que no usamos la variable `i_switch`
```suggestion
    for _ in range(switches):
```

devarops commented on mean_differences_mc/switch_elements.py:
```diff
@@ -1,18 +1,20 @@
-    return(A_final, B_final)
+    return (A_final, B_final)
+
 
```
En este archivo, los nombres de las variables deberían ser `a_final`, `b_final`, `A` y `B` para cumplir con PEP-8.

devarops reviewed (changes requested):
Aquí va mi primer revisión

Commit:
Atiende correcciones de Evaristo

Commit:
Agrega prueba para cazar mutantes 69 y 72

mutante 69 era del tipo Equality Operator
mutante 72 era del tipo Arithmetic Operator

Commit:
Da formato con black

Commit:
Agrega la prueba de la implementación

devarops committed:
Usa versión de Python más reciente

devarops committed:
Encapsula nombre del modulo en variable

Commit:
Merge branch 'feature/ejemplo_chapman' into feature/warm_up

Commit:
pasa prueba test_generalized_monte_carlo_test

devarops committed:
🔨 Actualiza phonies check, format y linter

devarops committed:
🚨 Desactiva warnings de Flake8

devarops commented:
En el archivo `TesterSignificantDifference.py` estamos importando `switch_elements()` que se encuentra en `switch_elements.py`. Sin embargo, en `TesterSignificantDifference.py` definimos `switch_elements()` por lo que terminamos ignorando la definición que importamos desde `switch_elements.py`.

devarops committed:
👷 Crea nuevo objetivo phony coverage

Commit:
agrega generalized_montecalo_test

Commit:
👾 Caza mutante en el mensaeje de error de generalized_monte_carlo_test

devarops committed:
♻️ Refactoriza eliminando else

Commit:
👾 Caza mutante con fernandiña

Commit:
📝 Cambia Dockerfile a versión 3.7

Commit:
📝 Mejora importaciones para pasar pylint

Commit:
🎨 refactoriza de acuerdo a las sugerencias de pylint

Commit:
:recycle: Refactoriza código repetido

Commit:
:recycle: Cambia nombre de variable
